int64[] list
int32 id
